### ISSUE TITLE
Fix "non-static initialization of a flexible array member" error in block/scsi_ioctl.c

### ISSUE THREAD

**Issue by GustavoARSilva** [KSPP/linux#46]
When replacing the one-element array *reserved[1]* with a flexible-array member in struct *compat_cdrom_generic_command*:

![image](https://user-images.githubusercontent.com/245005/81013531-e4fb6600-8e20-11ea-9a1b-5498d13fa28e.png)

the following error shows up:

![image](https://user-images.githubusercontent.com/245005/81013123-453dd800-8e20-11ea-8200-f714d85455d4.png)

See [flexible-array conversions](https://github.com/KSPP/linux/issues/21) issue.

**Comment by kees** [comment]
I think just convert that from reserved[1] to reserved and adjust code to address it without an array index.

I wonder if we could use this trick to rename it in uapi:

```
diff --git a/block/scsi_ioctl.c b/block/scsi_ioctl.c
index ef722f04f88a..ca8e94daaa9f 100644
--- a/block/scsi_ioctl.c
+++ b/block/scsi_ioctl.c
@@ -653,7 +653,7 @@ struct compat_cdrom_generic_command {
        unsigned char   data_direction;
        compat_int_t    quiet;
        compat_int_t    timeout;
-       compat_caddr_t  reserved[1];
+       compat_caddr_t  unused;
 };
 #endif
 
@@ -675,7 +675,7 @@ static int scsi_get_cdrom_generic_arg(struct cdrom_generic_command *cgc,
                        .data_direction = cgc32.data_direction,
                        .quiet          = cgc32.quiet,
                        .timeout        = cgc32.timeout,
-                       .reserved[0]    = compat_ptr(cgc32.reserved[0]),
+                       .unused         = compat_ptr(cgc32.unused),
                };
                memcpy(&cgc->cmd, &cgc32.cmd, CDROM_PACKET_SIZE);
                return 0;
diff --git a/include/uapi/linux/cdrom.h b/include/uapi/linux/cdrom.h
index 2817230148fd..275fc1c659c1 100644
--- a/include/uapi/linux/cdrom.h
+++ b/include/uapi/linux/cdrom.h
@@ -289,7 +289,10 @@ struct cdrom_generic_command
        unsigned char           data_direction;
        int                     quiet;
        int                     timeout;
-       void                    __user *reserved[1];    /* unused, actually */
+       union {
+               void            __user *reserved[1];    /* unused, actually */
+               void            __user *unused;
+       };
 };
 
 /*
```

**Comment by GustavoARSilva** [comment]
This patch has been taken by Jens Axboe and will show up in mainline, soon:

https://lore.kernel.org/linux-hardening/20201002231033.GA6273@embeddedor/

**Comment by kees** [comment]
Now landed in commit f5ace5ef37b1e1de49882248656f35c45e041585
